### ISSUE TITLE
Textually same but binary different

### ISSUE THREAD

**Issue by martinsson** [approvals/Approvals.NodeJS#42]
Getting a problem when copy pasting the received file into the approved one. The diff is still there even though the text is identical. 

reporters: Meld, Kdiff3 at least
os: Macos, Linux

I'll dig into this a bit deeper soon, but posting in case someone ran into the same thing
Johan

**Comment by isidore** [comment]
this is either from 1 of two causes (based on other systems)

1) Header. The empty file might not have the UTF-8 header. 
Solution: generate the empty file with a UTF-8 header

2) Trailing whitespace. The diff tool might run a trim() before saving
Solution: I think the solution was to end with an empty line. not sure I remember this right.

**Comment by staxmanade** [comment]
@martinsson I took a look at your change in https://github.com/martinsson/Approvals.NodeJS/commit/11925e673a4730ae05c9b2f631c6ed7cd36a052a and have some concerns/thoughts.

Line-endings have been a PITA, so any improvement we can make to help us (developers) out here is awesome.

We should avoid reporting something as "approved" if there are line-ending differences. So let's be careful about the normalized comparison.

It would be awesome if we did a "line-ending normalized diff" and if that reports a different result than a "non-normalized line-ending diff" - we raise the diff error with a message that

1. States that the only differences we detected were whitespace
2. Try to come up with some possible solutions.

The project has some default configs for setting specific line-endings. (maybe those are not working in all cases? - but hopefully we can prescribe a good approach in the error message raised)


Thoughts?
